app/src/sharedTest/java/org/oppia/app/recyclerview/BindableAdapterTest.kt

```diff
@@ -199,16 +215,83 @@ class BindableAdapterTest {
 
       // Verify that all three values are bound in the correct order and with the correct values.
       scenario.onActivity { activity ->
-        val recyclerView: RecyclerView = getTestFragment(activity).view!!.findViewById(R.id.test_recycler_view)
+        val recyclerView: RecyclerView =
+          getTestFragment(activity).view!!.findViewById(R.id.test_recycler_view)
         assertThat(recyclerView.childCount).isEqualTo(3)
       }
 
       onView(atPosition(R.id.test_recycler_view, 0)).check(matches(withText(STR_VALUE_1.strValue)))
-      onView(atPosition(R.id.test_recycler_view, 1)).check(matches(withSubstring(INT_VALUE_0.intValue.toString())))
-      onView(atPosition(R.id.test_recycler_view, 2)).check(matches(withSubstring(INT_VALUE_1.intValue.toString())))
+      onView(
+        atPosition(
+          R.id.test_recycler_view,
+          1
+        )
+      ).check(matches(withSubstring(INT_VALUE_0.intValue.toString())))
+      onView(
+        atPosition(
+          R.id.test_recycler_view,
+          2
+        )
+      ).check(matches(withSubstring(INT_VALUE_1.intValue.toString())))
     }
   }
 
+  @Test
+  fun testRecyclerDataDiffCallback_getOldListSize_correctListSize() {
+    val oldList = listOf(STR_VALUE_1, STR_VALUE_0, STR_VALUE_2).toMutableList()
+    val newList = listOf(STR_VALUE_1, STR_VALUE_0).toMutableList()
+    val diffObj = RecyclerDataDiffCallback(oldList, newList)
```

Comment:
When testing the new behaviors that DiffUtil introduces for the adapter, we shouldn't be testing diff util itself (it's actually an implementation detail). Can you update these tests to test the new behaviors that become available but only by interacting with BindableAdapter? Consider that in the future we could switch out our implementation away from DiffUtil to something else entirely. These tests ought to still pass with the new implementation without requiring changes if they are properly testing the public API of the system under test and not its implementation.

Reply:
I added test cases for testing `diffUtil` itself because I am comparing class types here and this is our side implementation which is different from the original behaviour of `diffUtil` itself. Though, I should add some test cases only by interacting with BindableAdapter.

Reply:
Added a test case, which checks the BindableAdapter weather the list updated with the new data or not. 
As per the future point you said, with this test case no matter whatever the implementation/logic which use to update the list, this test case will tell us the data is going to update or not.

Reply:
I left another comment in this review pass, but what I think we want are converting the tests you wrote earlier to be based on the actual recycler view object rather than testing the data diff callback. Is it possible for us to observe update notifications to the recycler view & assert based on those?

Reply:
Done

Reply:
Resolving this now on behalf of @BenHenning